v1/fcm.go

```
@@ -0,0 +1,99 @@
+	PrivateKey         string `json:"private_key"`
+	ClientEmail        string `json:"client_email"`
+	ClientId           string `json:"client_id"`
+	AuthUri            string `json:"auth_uri"`
+	TokenUri           string `json:"token_uri"`
+}
+
+type fcmClient struct {
+	httpClient           *http.Client
+	ServiceAccountConfig *ServiceAccount
+}
+
+func getServiceAccountFromContent(content string) (*ServiceAccount, error) {
+	serviceAccount := &ServiceAccount{}
+	err := json.Unmarshal([]byte(content), &serviceAccount)
+	if err != nil {
+		return nil, fmt.Errorf("error umarshalling service account file>%v", err)
+	}
+
+	return serviceAccount, nil
+}
+
+func NewFcmClient(serviceAccountFileContent string, ctx context.Context) (*fcmClient, error) {
+	serviceAccountConfig, err := getServiceAccountFromContent(serviceAccountFileContent)
+	if err != nil {
+		return nil, err
+	}
+
+	conf := &jwt.Config{
+		Email:        serviceAccountConfig.ClientEmail,
+		PrivateKeyID: serviceAccountConfig.PrivateKeyId,
+		PrivateKey:   []byte(serviceAccountConfig.PrivateKey),
+		Scopes:       []string{FirebaseRequestScope},
+		TokenURL:     google.JWTTokenURL,
+	}
+
+	return &fcmClient{
+		httpClient:           conf.Client(ctx),
+		ServiceAccountConfig: serviceAccountConfig,
+	}, nil
+}
+
+func (c *fcmClient) Send(m FcmMessageBody) (*FcmSendHttpResponse, error) {
+	sendURL := fmt.Sprintf("%s/projects/%s/messages:send",
+		BaseURL, c.ServiceAccountConfig.ProjectId,
+	)
+
+	body, err := json.Marshal(m)
+	if err != nil {
+		return nil, fmt.Errorf("error unmarshaling message>%v", err)
+	}
+
+	resp, err := c.httpClient.Post(sendURL, DefaultContentType, bytes.NewReader(body))
+	if err != nil {
+		return nil, fmt.Errorf("error sending request to HTTP connection server>%v", err)
+	}
+
+	responseBody, _ := io.ReadAll(resp.Body)
+	defer resp.Body.Close()
+
+	fcmResp := &FcmSendHttpResponse{
+		Status: resp.StatusCode,
+	}
+
+	if fcmResp.Status != http.StatusOK {
+		return fcmResp, nil
+	}
```

This should return an error. Right?